bitbots_vision_tools/scripts/colorpicker.py

```diff
@@ -0,0 +1,254 @@
+            "Click to select the colors in the OpenCV window.\n"
+            "Use <shift> and <left click> to remove colors.\n"
+            "To undo a step press 'u'.\n"
+            "To increase the size of the selection box press '+', to decrease press '-' \n"
+            "To save the color space press 's' followed by the path and <enter>.\n"
+            "Exit using <esc>.\n\n\n", logger_name="colorpicker")
+
+
+    def _image_callback(self, msg):
+        """
+        Image message callback.
+
+        :param msg: Image message
+        """
+        self._image = self._cv_bridge.imgmsg_to_cv2(msg, 'bgr8')
+
+    def ui_loop(self):
+        """
+        Runs the UI window.
+
+        :return: Does the user exit the loop with a save flag set?
+        """
+        save = False
+        while not rospy.is_shutdown():
+            # Copy image for the canvas
+            canvas = self._image.copy()
+            # Get the mouse coordinates
+            x, y = self._mouse_coord
+
+            # Get the values for the selection box
+            box_min_x = max(int(x - self._box_size / 2), 0)
+            box_min_y = max(int(y - self._box_size / 2), 0)
+            box_max_x = min(int(x + self._box_size / 2), self._image.shape[1] - 1)
+            box_max_y = min(int(y + self._box_size / 2), self._image.shape[0] - 1)
+
+            # Check for click event
+            if self._left_click or self._undo_click:
+                # Get pixels in the selction
+                selected_matrix = self._image[
+                                            box_min_y : box_max_y,
+                                            box_min_x : box_max_x
+                                        ]
+                # Serialize the colors
+                selected_matrix = selected_matrix.reshape(-1,3)
+                # Get the unique colors
+                selected_matrix = np.unique(selected_matrix, axis=0)
+
+                # Copy the latest color space
+                color_space = self._history[-1].copy()
+
+                # Check for left click or shift + left click
+                if self._undo_click:
+                    # Remove color values from color space
+                    color_space[
+                        selected_matrix[:,0],
+                        selected_matrix[:,1],
+                        selected_matrix[:,2]] = 0
+                # Check if we add stuff
+                elif self._left_click:
+                    # Add color values to color space
+                    color_space[
+                        selected_matrix[:,0],
+                        selected_matrix[:,1],
+                        selected_matrix[:,2]] = 1
+                else:
+                    print("This should never happen")
+
+                # Append new color space to self._history
+                self._history.append(color_space)
+
+                # Reset events
+                self._left_click, self._undo_click = False, False
```

Comment:
```suggestion
                self._left_click = self._undo_click = False
```

Reply:
These are independent states and this syntax is quite confusing.